buildSrc/src/main/groovy/org/elasticsearch/gradle/test/DistroTestPlugin.java

```diff
@@ -0,0 +1,286 @@
+        return new Object() {
+            @Override
+            public String toString() {
+                if (vagrant.isWindowsVM()) {
+                    return convertWindowsPath(project, jdk.toString()) + additionalWindows;
+                }
+                return convertLinuxPath(project, jdk.toString()) + additionaLinux;
+            }
+        };
+    }
+
+    private void configureCopyPackagingTask(Project project) {
+
+        // temporary, until we have tasks per distribution
+        TaskProvider<Copy> copyPackagingArchives = project.getTasks().register(COPY_PACKAGING_TASK, Copy.class);
+        copyPackagingArchives.configure(t -> {
+            Provider<Directory> archivesDir = project.getLayout().getBuildDirectory().dir("packaging/archives");
+            t.into(archivesDir);
+            t.from(project.getConfigurations().getByName(PACKAGING_DISTRIBUTION));
+            t.doLast(action -> {
+                try {
+                    Path archivesPath = archivesDir.get().getAsFile().toPath();
+                    Files.writeString(archivesPath.resolve("version"), VersionProperties.getElasticsearch());
+
+                    // write bwc version, and append -SNAPSHOT if it is an unreleased version
+                    ExtraPropertiesExtension extraProperties = project.getExtensions().getByType(ExtraPropertiesExtension.class);
+                    BwcVersions bwcVersions = (BwcVersions) extraProperties.get("bwcVersions");
+                    String version = upgradeVersion.toString();
+                    if (bwcVersions.unreleasedInfo(upgradeVersion) != null) {
+                        version += "-SNAPSHOT";
+                    }
+                    Files.writeString(archivesPath.resolve("upgrade_from_version"), version);
+                    // this is always true, but bats tests rely on it. It is just temporary until bats is removed.
+                    Files.writeString(archivesPath.resolve("upgrade_is_oss"), "");
+                } catch (IOException e) {
+                    throw new UncheckedIOException(e);
+                }
+            });
+        });
+    }
+
+    private static void configureDistroTest(Project project) {
+        project.getPluginManager().apply(JavaPlugin.class);
+        BuildPlugin.configureCompile(project);
+        BuildPlugin.configureRepositories(project);
+        BuildPlugin.configureTestTasks(project);
+        BuildPlugin.configureInputNormalization(project);
+
+        TaskProvider<Test> destructiveTest = project.getTasks().register("destructiveDistroTest", Test.class);
+        destructiveTest.configure(t -> {
+            t.setMaxParallelForks(1);
+            if (System.getProperty(IN_VM_SYSPROP) == null) {
+                t.dependsOn(COPY_PACKAGING_TASK);
+            }
+        });
+
+        // setup outer task to run
+        TaskProvider<GradleDistroTestTask> testCaller = project.getTasks().register("distroTest", GradleDistroTestTask.class);
```

Comment:
There's a method to register and configure in a single call 
https://docs.gradle.org/current/javadoc/org/gradle/api/tasks/TaskContainer.html#register-java.lang.String-java.lang.Class-org.gradle.api.Action- 

I think that would be much more readable than `.configure(` on another line.

Reply:
I tend to dislike chained lines and find them less readable.